packages/react-graphql/src/types.ts

```
@@ -58,4 +58,5 @@ export interface AsyncQueryComponentType<Data, Variables, DeepPartial>
   KeepFresh(
     props: VariableOptions<Variables> & {pollInterval?: number} & ConstantProps,
   ): React.ReactElement<{}>;
+  resolve(): Promise<DocumentNode<Data, Variables, DeepPartial> | null>;
```

Shouldn't this always resolve to a real document node?

---

not if the promise fail right?

---

If it fails you should throw an error, there isn't a very good recovery case.